packages/@aws-cdk/aws-ecs-patterns/test/ec2/test.l3s.ts

```diff
@@ -60,6 +61,102 @@ export = {
     test.done();
   },
 
+  'ApplicationLoadBalancedEc2Service desiredCount can be undefined when feature flag is set'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    stack.node.setContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT, true);
+
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new ecs.Cluster(stack, 'Cluster', { vpc });
+    cluster.addCapacity('DefaultAutoScalingGroup', { instanceType: new ec2.InstanceType('t2.micro') });
+
+    // WHEN
+    new ecsPatterns.ApplicationLoadBalancedEc2Service(stack, 'Service', {
+      cluster,
+      memoryLimitMiB: 1024,
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry('test'),
+      },
+    });
+
+    expect(stack).notTo(haveResource('AWS::ECS::Service', {
+      DesiredCount: 1,
+    }));
```

Comment:
Nit: It might be nice to explicitly assert that the resource is there but does not contain a specific property, e.g.:
```suggestion
    expect(stack).to(haveResource('AWS::ECS::Service', {
      DesiredCount: ABSENT,
    }));
```
`ABSENT` can be imported from the assert lib. Same with all the assertions below.

Reply:
The ec2 module actually uses both to and notTo, rather than ABSENT. I was trying to stick to what's already there @MrArnoldPalmer what's the preferred approach here?

Reply:
For asserting that a specific property is undefined, the usage of `ABSENT` is better. Not having a resource with `desiredCount: 1` means that it could have a resource with `desiredCount: 2`